indexinglanguage/src/test/java/com/yahoo/vespa/indexinglanguage/ScriptTestCase.java

```diff
@@ -99,6 +98,52 @@ public void testLiteralBoolean() throws ParseException {
         assertEquals(new BoolFieldValue(true), output.getFieldValue("mybool"));
     }
 
+    @Test
+    public void testIntHash() throws ParseException {
+        var expression = Expression.fromString("input myText | hash | attribute 'myInt'");
+
+        SimpleTestAdapter adapter = new SimpleTestAdapter();
+        adapter.createField(new Field("myText", DataType.STRING));
+        var intField = new Field("myInt", DataType.INT);
+        adapter.createField(intField);
+        adapter.setValue("myText", new StringFieldValue("input text"));
+        expression.setStatementOutput(new DocumentType("myDocument"), intField);
+
+        // Necessary to resolve output type
+        VerificationContext verificationContext = new VerificationContext(adapter);
+        assertEquals(DataType.INT, expression.verify(verificationContext));
+
+        ExecutionContext context = new ExecutionContext(adapter);
+        context.setValue(new StringFieldValue("input text"));
+        expression.execute(context);
+        assertNotNull(context);
+        assertTrue(adapter.values.containsKey("myInt"));
+        assertEquals(-1425622096, adapter.values.get("myInt").getWrappedValue());
+    }
+
+    @Test
+    public void testLongHash() throws ParseException {
+        var expression = Expression.fromString("input myText | hash | attribute 'myLong'");
+
+        SimpleTestAdapter adapter = new SimpleTestAdapter();
+        adapter.createField(new Field("myText", DataType.STRING));
+        var intField = new Field("myLong", DataType.LONG);
+        adapter.createField(intField);
+        adapter.setValue("myText", new StringFieldValue("input text"));
+        expression.setStatementOutput(new DocumentType("myDocument"), intField);
+
+        // Necessary to resolve output type
+        VerificationContext verificationContext = new VerificationContext(adapter);
+        assertEquals(DataType.LONG, expression.verify(verificationContext));
+
+        ExecutionContext context = new ExecutionContext(adapter);
+        context.setValue(new StringFieldValue("input text"));
+        expression.execute(context);
```

Comment:
The setup and execution is identical to that used in testIntHash(), except the data type and field name. Consider refactor into a helper function. As it stands I think it is harder to read as I was looking for differences between the two tests.